### ISSUE TITLE
Allow overriding retreival of EndUserIp

### ISSUE THREAD

**Issue by PeterOrneholm** [ActiveLogin/ActiveLogin.Authentication#221]
**Is your feature request related to a problem? Please describe.**
We have had multiple issues (#152, #213) related to what we use as a value for EndUserIp. Seems that our current solution `HttpContext.Connection.RemoteIpAddress.ToString()` might be too naive in scenarios with Docker, Proxies etc.

**What area is it related to**
- ActiveLogin.Authentication.BankId.AspNetCore

**Describe the solution you'd like**
Add an interface that has one method that should return the value to use for EndUserIp. The default implementation implemented in #215 but can be overridden depending on the scenario. We should also add extension methods to make that easier.

Suggestion:

An implementation could take whatever dependency it needs, but as HttpContext will be used in 99.9% of the implementations, that's what we will send to the method.

```csharp
public interface IBankIdEndUserIpResolver {
      string GetEndUserIp(HttpContext context);
}

public static class BankIdBuilderExtensions
    {
        public static IBankIdBuilder UseEndUserIpResolver<TEndUserIpResolver>(this IBankIdBuilder builder) where TEndUserIpResolver : IBankIdEndUserIpResolver
        {
            ...
        }

        public static IBankIdBuilder UseEndUserIpResolver(this IBankIdBuilder builder, Func<HttpContext, string> configureEndUserIpResolver)
        {
            ...
        }
    }
```

**Comment by PeterOrneholm** [comment]
You both have had issues with end user ip @span and @magohl. What do you think about this solution?

**Comment by magohl** [comment]
For me X-Forwarded-For headers (using the UseForwardedHeaders middleware) together with the fix in #213 has worked well hosted in docker/linux with dual (!) chained proxies in front.

But I think it is a very good idea to allow overriding the default behavior as suggested. Who knows when a new standard emerges related to proxies, headers or similar.

**Comment by span** [comment]
I think it would be a great addition!
